### PR TITLE
chore: adding missing translation function

### PR DESCRIPTION
**Pull request by AndVK**
### ADDITIONAL INFORMATION

- [ ] Has associated issue: Fixes https://github.com/apache/superset/issues/17779
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/components/ListView/Filters/DateRange.tsx [hunk 1]
```diff
@@ -62,7 +62,7 @@ function DateRangeFilter(
 
   return (
     <RangeFilterContainer>
-      <FormLabel>{Header}</FormLabel>
+      <FormLabel>{t(Header as string)}</FormLabel>
```
Comment (rusackas) [inline superset-frontend/src/components/ListView/Filters/DateRange.tsx:65]: Can you explain the "as string" need here? If this is a dynamic value, it might not be wise to use the t() here.

File: superset-frontend/src/components/TableLoader/index.tsx [hunk 2]
```diff
@@ -66,9 +66,9 @@ const TableLoader = (props: TableLoaderProps) => {
       tableColumns = Object.keys(data[0]).filter(col => col[0] !== '_');
     }
     return tableColumns
-      ? tableColumns.map((column: string) => ({
-          accessor: column,
-          Header: column,
+      ? tableColumns.map((column: string[] | string) => ({
+        accessor: column[0],
+        Header: column[1],
```
Comment (rusackas) [inline superset-frontend/src/components/TableLoader/index.tsx:71]: Is this bycatch? Might need a little explanation of why this is tackled in this PR.

File: superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterLabel.tsx [hunk 3]
```diff
@@ -318,7 +318,7 @@ export default function DateFilterLabel(props: DateFilterControlProps) {
         {!validTimeRange && (
           <IconWrapper className="warning">
             <Icons.ErrorSolidSmall iconColor={theme.colors.error.base} />
-            <span className="text error">{evalResponse}</span>
+            <span className="text error">{t(evalResponse)}</span>
```
Comment (rusackas) [inline superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterLabel.tsx:321]: I don't know what strings would be expected here, but this sounds highly dynamic and unlikely to be in the translation files. Correct me if I'm mistaken.

File: superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterLabel.tsx [hunk 4]
```diff
@@ -373,7 +373,7 @@ export default function DateFilterLabel(props: DateFilterControlProps) {
       >
         <Tooltip placement="top" title={tooltipTitle}>
           <Label className="pointer" data-test="time-range-trigger">
-            {actualTimeRange}
+          {t(`${actualTimeRange}`)}
```
Comment (rusackas) [inline superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterLabel.tsx:376]: I don't know what strings would be expected here, but this sounds highly dynamic and unlikely to be in the translation files. Correct me if I'm mistaken.

File: superset-frontend/src/profile/components/CreatedContent.tsx [hunk 5]
```diff
@@ -53,7 +56,7 @@ class CreatedContent extends React.PureComponent<CreatedContentProps> {
     const search = [{ col: 'created_by', opr: 'created_by_me', value: 'me' }];
     const query = rison.encode({
       keys: ['none'],
-      columns: ['created_on_delta_humanized', 'dashboard_title', 'url'],
+      columns: ['changed_on_utc', 'dashboard_title', 'url'],
```
Comment (rusackas) [inline superset-frontend/src/profile/components/CreatedContent.tsx:59]: Please add context on this change... doesn't seem related to the general translation theme of the PR.

File: superset-frontend/src/profile/components/CreatedContent.tsx [hunk 6]
```diff
@@ -63,8 +66,8 @@ class CreatedContent extends React.PureComponent<CreatedContentProps> {
     const mutator = (data: DashboardResponse) =>
       data.result.map(dash => ({
         dashboard: <a href={dash.url}>{dash.dashboard_title}</a>,
-        created: dash.created_on_delta_humanized,
-        _created: dash.created_on_delta_humanized,
+        created: moment.utc(dash.changed_on_utc).fromNow(),
```
Comment (rusackas) [inline superset-frontend/src/profile/components/CreatedContent.tsx:69]: Again, more context here would be helpful.

File: superset-frontend/src/profile/components/Favorites.tsx [hunk 7]
```diff
@@ -64,16 +68,19 @@ export default class Favorites extends React.PureComponent<FavoritesProps> {
     const mutator = (data: DashboardResponse) =>
       data.result.map(dash => ({
         dashboard: <a href={dash.url}>{dash.dashboard_title}</a>,
-        created: dash.created_on_delta_humanized,
-        _created: dash.created_on_delta_humanized,
+        created: moment.utc(dash.changed_on_utc).fromNow(),
```
Comment (rusackas) [inline superset-frontend/src/profile/components/Favorites.tsx:71]: Context for this change also welcomed, thanks!

File: superset-frontend/src/views/components/LanguagePicker.tsx [hunk 8]
```diff
@@ -76,7 +76,7 @@ export default function LanguagePicker(props: LanguagePickerProps) {
         >
           <StyledLabel className="f16">
             <i className={`flag ${languages[langKey].flag}`} />
-            <a href={languages[langKey].url}>{languages[langKey].name}</a>
+            <a href={languages[langKey].url}>{t(languages[langKey].name)}</a>
```
Comment (rusackas) [inline superset-frontend/src/views/components/LanguagePicker.tsx:79]: Just to make sure, I hope having these translated doesn't cause an issue with people having a harder time switching to their preferred language!

**Comment by rusackas** [comment]
Most of this seems a-ok! Thanks for the contribution! The vast majority of the `t()` calls seem unobjctionable.

There is, however, a fair amount of change in this PR involving Moment and utc timestamps. A description of how that fits in here (and any PR description at all, for that matter) would be helpful. 

There are also a couple of translated strings in there that look like they might be fairly dynamic in terms of content. We shouldn't use the t() wrapper unless the translated string is static/stable.

Also noting that this should close out https://github.com/apache/superset/issues/21200 and this can be done automatically by adding "Fixes: #21200" in your PR description.

**Comment by rusackas** [comment]
Hi @AndVK :) I haven't forgotten about this... any chance this can get a rebase and we can push it through? It's _so close_

**Comment by rusackas** [comment]
@AndVK if this is getting to painful to rebase until CI passes and so forth, I'm also fine with splitting it up into smaller PRs that require less strenuous reviews ;) Either way, I still appreciate all this great/hard work!

**Comment by AndVK** [comment]
Hi, @rusackas! I will split this PR into smaller ones.

**Comment by rusackas** [comment]
> Hi, @rusackas! I will split this PR into smaller ones.

Fantastic, looking forward to it! Ping me/us here and/or on Slack to make sure we pay attention to them! :D

**Comment by rusackas** [comment]
@AndVK @hbruch do either of you have the bandwidth to carry any pieces of this forward? Leaving it open since it serves as a reminder, and hoping that we can close out #21200 sooner rather than later 🤞

**Comment by hbruch** [comment]
Would you mind reviewing https://github.com/apache/superset/pull/23402 before?

**Comment by hbruch** [comment]
I went through the suggested changes and most of them were already handled and merged via @AndVK's https://github.com/apache/superset/pull/21816 (thanks again!). For the very few missing translations I spotted, I'll submit a PR.

Besides this, two changes suggested by @AndVK are still open and require some code changes:
* internationalize [table columns](https://github.com/apache/superset/blob/master/superset-frontend/src/components/TableLoader/index.tsx#L68-L71)
* humanized times/ time deltas. For these, I also think, they should be handled client side

Another category of missing translations are error messages. Not sure, which of the are really end user facing, but a couple of them them will be (e.g. in [controls.ts](https://github.com/apache/superset/blob/master/superset-frontend/plugins/plugin-chart-echarts/src/utils/controls.ts#L35) or [TTestTable.jsx](https://github.com/apache/superset/blob/master/superset-frontend/plugins/legacy-plugin-chart-paired-t-test/src/TTestTable.jsx#L176)).

**Comment by rusackas** [comment]
Thanks @hbruch. Closing this PR for now, and appreciate the follow-up!
